src/main/java/it/smartcommunitylab/aac/saml/provider/SamlAuthenticationProvider.java

```diff
@@ -227,6 +240,18 @@ protected SamlUserAuthenticatedPrincipal createUserPrincipal(Object principal) {
         // we still don't have userId
         String userId = null;
 
+        // subjectId is optionally evaluated from an attribute defined in provider configurations
+        String subjectId = samlDetails.getName();
+        String subAttributeName = config.getConfigMap().getSubAttributeName();
```

Comment:
set this to a local variable when the constructor is executed
```
 // let config set subject attribute name
        this.subAttributeName = config.getSubAttributeName();
```

also the value should be read from Config, not from the ConfigMap.

Reply:
Ok for using a class attribute evaluated in the constructor.

Is there a specific reason as to why it is preferred to read from Config over ConfigMap? Values are read directly from the ConfigMap in other places of the same class, like [here in the constructor](https://github.com/scc-digitalhub/AAC/blob/895984a76b5fbf9298fed70dec48d640eee700fc/src/main/java/it/smartcommunitylab/aac/saml/provider/SamlAuthenticationProvider.java#L95) or [here in the same method](https://github.com/scc-digitalhub/AAC/blob/895984a76b5fbf9298fed70dec48d640eee700fc/src/main/java/it/smartcommunitylab/aac/saml/provider/SamlAuthenticationProvider.java#L287)